backends/tfhe-cuda-backend/cuda/src/integer/scalar_comparison.cuh

```diff
@@ -2,16 +2,44 @@
 #define CUDA_INTEGER_SCALAR_COMPARISON_OPS_CUH
 
 #include "integer/comparison.cuh"
+template <typename Torus>
+Torus is_x_less_than_y_given_input_borrow(Torus last_x_block,
+                                          Torus last_y_block, Torus borrow,
+                                          uint32_t message_modulus) {
+  Torus last_bit_pos = log2_int(message_modulus) - 1;
+  Torus mask = (1 << last_bit_pos) - 1;
+  Torus x_without_last_bit = last_x_block & mask;
+  Torus y_without_last_bit = last_y_block & mask;
+
+  bool input_borrow_to_last_bit =
+      x_without_last_bit < (y_without_last_bit + borrow);
+
+  Torus result = last_x_block - (last_y_block + borrow);
+
+  Torus output_sign_bit = (result >> last_bit_pos) & 1;
```

Comment:
Is `Torus` correct return type,  values function can return is `0` or `1`

Reply:
On the Rust side it is defined as Torus, so I did the same.